### PR TITLE
Tta #61 add try catches to functions

### PR DESCRIPTION
**Pull request by mnmin**

### CODE REVIEW

File: prisma/migrations/20221002205940_add_excercise_and_event_type_enum/migration.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+  - Added the required column `updatedAt` to the `Exercise` table without a default value. This is not possible if the table is not empty.
+
+*/
+-- AlterEnum
+ALTER TYPE "EventType" ADD VALUE 'EXERCISE';
+
+-- AlterTable
+ALTER TABLE "Exercise" ADD COLUMN     "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+ADD COLUMN     "updatedAt" TIMESTAMP(3) NOT NULL;
```
Comment (vherus) [inline prisma/migrations/20221002205940_add_excercise_and_event_type_enum/migration.sql:12]: This migration will crash the production database. Please delete this file, add a default to the `updatedAt` column on the Exercise model of `@default(now())` and generate a new migration

File: prisma/migrations/20221004074643_add_createdat_and_updatedAt_to_cohort/migration.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+-- AlterTable
+ALTER TABLE "Cohort" ADD COLUMN     "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+ADD COLUMN     "updatedAt" TIMESTAMP(3);
```
Comment (vherus) [inline prisma/migrations/20221004074643_add_createdat_and_updatedAt_to_cohort/migration.sql:3]: This migration will crash the production database. Please delete this file, add a default to the `updatedAt` column on the Cohort model of `@default(now())` and generate a new migration

**Review by vherus — CHANGES_REQUESTED** [review]
Nice. Just a couple of fixes to make here

File: prisma/migrations/20221002205940_add_excercise_and_event_type_enum/migration.sql (hunk 1 above)
Comment (mnmin) [reply prisma/migrations/20221002205940_add_excercise_and_event_type_enum/migration.sql]: Updated

File: prisma/migrations/20221004074643_add_createdat_and_updatedAt_to_cohort/migration.sql (hunk 2 above)
Comment (mnmin) [reply prisma/migrations/20221004074643_add_createdat_and_updatedAt_to_cohort/migration.sql]: Updated

File: prisma/schema.prisma [hunk 3]
```diff
@@ -58,6 +58,8 @@ model Cohort {
   deliveryLogs DeliveryLog[]
   name         String?       @unique
   event        Event[]
+  createdAt    DateTime      @default(now())
+  updatedAt    DateTime?     @default(now())
```
Comment (vherus) [inline prisma/schema.prisma:62]: `updatedAt DateTime? @updatedAt`

No need to add the `@default` if you do this

File: prisma/migrations/20221004094306_add_updated_at_default_now_cohort/migration.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+-- AlterTable
+ALTER TABLE "Cohort" ADD COLUMN     "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+ADD COLUMN     "updatedAt" TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP;
```
Comment (vherus) [inline prisma/migrations/20221004094306_add_updated_at_default_now_cohort/migration.sql:3]: Delete this file, not just the contents of it, and generate new migrations

File: prisma/schema.prisma [hunk 5]
```diff
@@ -135,13 +135,15 @@ model Exercise {
   objectives String[]
   users      ExerciseUser[]
   logs       DeliveryLog[]
+  createdAt  DateTime       @default(now())
+  updatedAt  DateTime       @default(now())
```
Comment (vherus) [inline prisma/schema.prisma:139]: `updatedAt DateTime? @updatedAt`

No need to add the @default if you do this

File: prisma/migrations/20221003142638_delivery_log_on_delete_casade/migration.sql [hunk 6]
```diff
@@ -0,0 +1 @@
+-- This is an empty migration.
```
Comment (vherus) [inline prisma/migrations/20221003142638_delivery_log_on_delete_casade/migration.sql:1]: Delete this file

**Review by vherus — CHANGES_REQUESTED** [review]
Almost

File: prisma/migrations/20221002205940_add_excercise_and_event_type_enum/migration.sql (hunk 1 above)
Comment (mnmin) [reply prisma/migrations/20221002205940_add_excercise_and_event_type_enum/migration.sql]: updated

File: prisma/schema.prisma (hunk 3 above)
Comment (mnmin) [reply prisma/schema.prisma]: updated

File: prisma/schema.prisma (hunk 5 above)
Comment (mnmin) [reply prisma/schema.prisma]: updated

File: prisma/migrations/20221003142638_delivery_log_on_delete_casade/migration.sql (hunk 6 above)
Comment (mnmin) [reply prisma/migrations/20221003142638_delivery_log_on_delete_casade/migration.sql:1]: deleted

**Review by vherus — APPROVED** [review]

**Review by Noemi890 — APPROVED** [review]

**Pull request merged**